### DISCUSSION TITLE
Docs: Is it possible to use filesystem storage in monolith mode cluster?

### DISCUSSION THREAD

**Discussion by 4orty** [grafana/mimir#1724] [Help and support] [answered]
Is it possible to use filesystem stroage in monolith mode cluster?
All tutorials uses minio in cluster and none of the documents says it's not working.

**Comment by pracucci** [comment, marked as answer]
> Is it possible to use filesystem stroage in monolith mode cluster?

Yes. Keep in mind that `filesystem` backend storage is primarily targeted for testing, when you only run 1 instance of Mimir.

Have you encountered any issue using it with the monolithic mode?

**Reply by 4orty** [reply]
Thanks for your rapid answer! 
I don't encountered any issue i just read docs before setup mimir.

A few more questions,
If i first use 1 instance and filesystem, is it easy to migrate from filesystem to object storage(minio) ??

And suddenly i felt that something weird in my question. 
'monolithic mode cluster' is it right statement? 
I'm confused about 'cluster' and 'monolithic / microservice' mode.
If i set up three mimir instance like [docs](https://grafana.com/docs/mimir/latest/operators-guide/architecture/deployment-modes/#monolithic-mode), is it possible to say it's a 'cluster' ?? In other words, in monolithic mode, is it possible to configure memberlist?

**Reply by pracucci** [reply]
> If i first use 1 instance and filesystem, is it easy to migrate from filesystem to object storage(minio) ??

Yes. You just need to upload blocks stored on local filesystem to the object storage.

> 'monolithic mode cluster' is it right statement?
> I'm confused about 'cluster' and 'monolithic / microservice' mode.

Monolithic mode means all Mimir components run within the same process. You can still horizontally scale the monolithic mode, running multiple replicas. However, local filesystem is just supported when you run monolithic mode with only 1 replica. To scale out, you will need to move to an object storage.

> If i set up three mimir instance like [docs](https://grafana.com/docs/mimir/latest/operators-guide/architecture/deployment-modes/#monolithic-mode), is it possible to say it's a 'cluster' ?? In other words, in monolithic mode, is it possible to configure memberlist?

Yes, absolutely!

**Reply by 4orty** [reply]
@pracucci
Thanks for your kind reply, i have a better understanding of mimir.

+ I'm so sorry to bother you 😭 but 

> Yes. You just need to upload blocks stored on local filesystem to the object storage.

Can you simply share how to upload blocks from local filesystem to object storage? 
I read [migrating-from-thanos-or-prometheus](https://grafana.com/docs/mimir/latest/migration-guide/migrating-from-thanos-or-prometheus/) but in this way, i think recent metrics before created to blocks(in prometheus, default 2 hours) will be lost..
